File: TAM.LogisticSystem/Services/MasterWarnaVehicleService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Dapper;
using Microsoft.EntityFrameworkCore;
using TAM.LogisticSystem.Entities;
using TAM.LogisticSystem.Models;

namespace TAM.LogisticSystem.Services
{
    public class MasterWarnaVehicleService
    {
        private readonly LogisticDbContext LogisticDbContext;

        public MasterWarnaVehicleService(LogisticDbContext logisticDbContext)
        {
            this.LogisticDbContext = logisticDbContext;
        }

        // TIE: START
        ///// <summary>
        ///// Add Vehicle Color to DB
        ///// </summary>
        ///// <param name="vehicleColorViewModel"></param>
        ///// <returns></returns>
        //public async Task AddNewVehicleColor(MasterWarnaVehicleCreateModel vehicleColorViewModel)
        //{
        //    var carTypeColor = new CarTypeColor
        //    {
        //        CarTypeColorCode = vehicleColorViewModel.KodeWarnaVehicle,
        //        CarModelCode = vehicleColorViewModel.Model.KodeModel,
        //        BrandCode = vehicleColorViewModel.Brand.KodeBrand,
        //        ExteriorColorCode = vehicleColorViewModel.Warna.KodeWarna
        //    };
        //    LogisticDbContext.CarTypeColor.Add(carTypeColor);
        //    await LogisticDbContext.SaveChangesAsync();
        //}
        ///// <summary>
        ///// Update Vehicle Color by KodeWarnaVehicle
        ///// </summary>
        ///// <param name="vehicleColorViewModel"></param>
        ///// <returns></returns>
        //public async Task UpdateVehicleColor(MasterWarnaVehicleCreateModel vehicleColorViewModel)
        //{
        //    var vehicleColor = await LogisticDbContext.CarTypeColor
        //        .Where(Q => Q.CarTypeColorCode == vehicleColorViewModel.KodeWarnaVehicle)
        //        .FirstOrDefaultAsync();

        //    vehicleColor.CarTypeColorCode = vehicleColorViewModel.KodeWarnaVehicle;
        //    vehicleColor.CarModelCode = vehicleColorViewModel.Model.KodeModel;
        //    vehicleColor.BrandCode = vehicleColorViewModel.Brand.KodeBrand;
        //    vehicleColor.ExteriorColorCode = vehicleColorViewModel.Warna.KodeWarna;

        //    LogisticDbContext.CarTypeColor.Update(vehicleColor);
        //    await LogisticDbContext.SaveChangesAsync();
        //}
        ///// <summary>
        ///// Delete Vehicle Color by KodeWarnaVehicle
        ///// </summary>
        ///// <param name="code"></param>
        ///// <returns></returns>
        //public async Task<int> RemoveVehicleColor(string code)
        //{
        //    var vehicleColor = await LogisticDbContext.CarTypeColor
        //        .Where(Q => Q.CarTypeColorCode == code)
        //        .FirstOrDefaultAsync();
        //    var result = LogisticDbContext.CarTypeColor.Remove(vehicleColor);
        //    return await LogisticDbContext.SaveChangesAsync();
        //}
        // TIE: END

        /// <summary>
        /// Get All Vehicle Color Data from DB
        /// </summary>
        /// <returns></returns>
        public async Task<List<MasterWarnaVehicleViewModel>> GetAllVehicleColor()
        {
            var con = LogisticDbContext.Database.GetDbConnection();
            {
                var vehicleColor = (await con.QueryAsync<MasterWarnaVehicleViewModel>(@"
	SELECT
	ctc.CarTypeColorCode AS [KodeWarnaVehicle],
	ctc.BrandCode AS [KodeBrand],
	b.Name AS [BrandName],
	ctc.CarModelCode AS [KodeModel],
	cm.Name AS [ModelName],
	ctc.ExteriorColorCode AS [KodeWarna],
	ec.IndonesianName AS [DeskripsiWarnaInd],
	ec.EnglishName AS [DeskripsiWarnaEng]	
FROM CarTypeColor ctc
	JOIN ExteriorColor ec ON ctc.ExteriorColorCode = ec.ExteriorColorCode
	JOIN Brand b ON ctc.BrandCode = b.BrandCode
	JOIN CarModel cm ON ctc.CarModelCode = cm.CarModelCode
                ")).ToList();

                if (vehicleColor != null)
                {
                    foreach (var item in vehicleColor)
                    {
                        item.BrandDetail = item.KodeBrand + " - " + item.BrandName;
                        item.ModelDetail = item.KodeModel + " - " + item.ModelName;
                    }
                }

               

                return vehicleColor;
            }
        }

        /// <summary>
        /// Get All Brand Data from DB
        /// </summary>
        /// <returns></returns>
        public async Task<List<MasterWarnaVehicleBrandModel>> GetAllKodeBrand()
        {
            var con = LogisticDbContext.Database.GetDbConnection();
            {
                var brandModel = (await con.QueryAsync<MasterWarnaVehicleBrandModel>(@"
                    SELECT  
	b.BrandCode AS [KodeBrand],
    b.Name AS [BrandName]
FROM Brand b
                ")).ToList();
                return brandModel;
            }
        }
        /// <summary>
        /// Get All Model Data from DB
        /// </summary>
        /// <returns></returns>
        public async Task<List<MasterWarnaVehicleModelModel>> GetAllKodeModel()
        {
            var con = LogisticDbContext.Database.GetDbConnection();
            {
                var modelModel = (await con.QueryAsync<MasterWarnaVehicleModelModel>(@"
                    SELECT  
	cm.CarModelCode AS [KodeModel],
	cm.BrandCode AS [KodeBrand],
    cm.Name AS [ModelName]
FROM CarModel cm
                ")).ToList();
                return modelModel;
            }
        }
        /// <summary>
        /// Get All Color Data from DB
        /// </summary>
        /// <returns></returns>
        public async Task<List<MasterWarnaVehicleColorModel>> GetAllKodeWarna()
        {
            var con = LogisticDbContext.Database.GetDbConnection();
            {
                var colorModel = (await con.QueryAsync<MasterWarnaVehicleColorModel>(@"
                    SELECT
	ec.ExteriorColorCode AS [KodeWarna],
	ec.IndonesianName AS [DeskripsiWarnaInd],
	ec.EnglishName AS [DeskripsiWarnaEng]
FROM ExteriorColor ec
                ")).ToList();
                return colorModel;
            }
        }

        // TIE: START
        ///// <summary>
        ///// Check if Vehicle Color already exists
        ///// </summary>
        ///// <param name="kodeWarnaVehicle"></param>
        ///// <returns></returns>
        //public async Task<bool> IsVehicleColorExist(string kodeWarnaVehicle)
        //{
        //    var data = await LogisticDbContext.CarTypeColor.FirstOrDefaultAsync(Q => Q.CarTypeColorCode == kodeWarnaVehicle);
        //    if (data != null)
        //    {
        //        return true;
        //    }
        //    return false;
        //}
        // TIE: END
    }
}

File: TAM.LogisticSystem/client/js/Services/MasterWarnaVehicleService.ts


export class MasterWarnaVehicleService {
    static $inject = ['$http'];

    httpClient: angular.IHttpService;

    constructor($http: angular.IHttpService) {
        this.httpClient = $http;
    }
    //get all data location type
    getAllVehicleColor() {
        return this.httpClient.get<MasterWarnaVehicleViewModel[]>('/api/v1/MasterWarnaVehicleApi');
    }

    getAllKodeBrand() {
        return this.httpClient.get<MasterWarnaVehicleBrandModel[]>('/api/v1/MasterWarnaVehicleApi/GetAllKodeBrand');
    }

    getAllKodeModel() {
        return this.httpClient.get<MasterWarnaVehicleModelModel[]>('/api/v1/MasterWarnaVehicleApi/GetAllKodeModel');
    }

    getAllKodeWarna() {
        return this.httpClient.get<MasterWarnaVehicleColorModel[]>('/api/v1/MasterWarnaVehicleApi/GetAllKodeWarna');
    }

    //add new location type
    postVehicleColor(vehicleColorModel: MasterWarnaVehicleCreateModel) {
        return this.httpClient.post('/api/v1/MasterWarnaVehicleApi', vehicleColorModel);
    }
    //udpate location type
    updateVehicleColor(vehicleColorModel: MasterWarnaVehicleCreateModel) {
        return this.httpClient.post('/api/v1/MasterWarnaVehicleApi/edit/', vehicleColorModel);
    }
    //delete location type
    deleteVehicleColor(kodeWarnaVehicle: string) {
        return this.httpClient.delete('/api/v1/MasterWarnaVehicleApi/delete/' + kodeWarnaVehicle);
    }
}

export class MasterWarnaVehicleViewModel {
    kodeWarnaVehicle: string;
    kodeBrand: string;
    brandName: string;
    brandDetail: string;
    kodeModel: string;
    modelName: string;
    modelDetail: string;
    kodeWarna: string;
    deskripsiWarnaInd: string;
    deskripsiWarnaEng: string;
}

export class MasterWarnaVehicleCreateModel {
    kodeWarnaVehicle: string;
    brand: MasterWarnaVehicleBrandModel;
    model: MasterWarnaVehicleModelModel;
    warna: MasterWarnaVehicleColorModel;
}

export class MasterWarnaVehicleBrandModel {
    kodeBrand: string;
    brandName: string;
}

export class MasterWarnaVehicleModelModel {
    kodeBrand: string;
    kodeModel: string;
    modelName: string;
}

export class MasterWarnaVehicleColorModel {
    kodeWarna: string;
    deskripsiWarnaInd: string;
    deskripsiWarnaEng: string;
}
